qemu/tests/blockdev_snapshot_guest_agent.py

```diff
@@ -63,8 +63,9 @@ def run(test, params, env):
     :param env: Dictionary with test environment.
     """
     base_image = params.get("images", "image1").split()[0]
-    params.update(
-        {"image_name_%s" % base_image: params["image_name"],
-         "image_format_%s" % base_image: params["image_format"]})
+    if not params.get("image_name_%s" % base_image):
+        params.update(
+            {"image_name_%s" % base_image: params["image_name"],
+             "image_format_%s" % base_image: params["image_format"]})
```

Comment:
image_format_<base_image> should be always set, no matter if image_name_<base_image> is set or not.
e.g. consider when the imageformat=raw, then we'll get a wrong backing image

also, you may try 
params.setdefault("image_name_%s" % base_image, params["image_name"])
params.setdefalut("image_format_%s" % base_image, params["image_format"])

Reply:
@zhencliu good idea, have updated the patch as your sugguestion, please help to review again, thanks.